media.go

```diff
@@ -433,6 +465,9 @@ func (ms *mediaStream[T, U]) Next(ctx context.Context) (T, func(), error) {
 		}
 	}
 
+	ctx, prodConLockSpan := trace.StartSpan(ctx, "gostream::mediaStream::Next (waiting for ms.prodCon lock)")
+	defer prodConLockSpan.End()
+
```

Comment:
You could also add a span to `Close` on line 483. I also have the same thoughts about nesting `stopOne` in `Close` the same way `stop` should be nested in `stopOne`

Reply:
[Here's a trace that shows the Close -> stopOne -> stop hierarchy that you were looking for](https://console.cloud.google.com/traces/list?project=av-canary&tid=cfd45d1add23bdd55b638687640e2417&pageState=(%22traceIntervalPicker%22:(%22groupValue%22:%22PT1H%22,%22customValue%22:null))&start=1689862283893&end=1689865883893). I tried to make sure that preemption wouldn't be affected by using the original cancelCtx instead of creating a new context outright. I tested that preemption is preserved by running the same test I used to generate the trace and halting execution with a sigint, and it looked like execution stopped as quickly as it did without the changes.